support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+// ----- Imports ----- //
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import { fetchJson } from 'helpers/fetch';
+import { logException } from '../logger';
+import { DirectDebit, ExistingCard, ExistingDirectDebit, Stripe } from '../paymentMethods';
+import { getPaymentLabel } from '../checkouts';
+
+// ----- Types ----- //
+
+export type ExistingPaymentMethodSubscription = {
+  isActive: boolean;
+  isCancelled: boolean,
+  name: string
+}
+
+type ExistingPaymentType = 'Card' | 'DirectDebit';
+
+export type PartialExistingPaymentMethod = {
+  paymentType: ExistingPaymentType;
+};
+
+export type FullDetailExistingPaymentMethod = {
```

Comment:
under what conditions are full details returned from the endpoint as opposed to partial? and could we put this information in a comment above the type/encode it in the naming?

Reply:
depending on the freshness of the sign-in, see in PR description...
> If the user is signed-in but not in the last hour (but user does have existing payment methods available)...
... the name could be better though, along with a comment... will do